### DISCUSSION TITLE
useLazyAsyncData refetching data after hydration after 3.6.2 in the client

### DISCUSSION THREAD

**Discussion by maxarias-io** [nuxt/nuxt#21993] [Questions]
After updating to 3.6.2, I'm seeing "useLazyAsyncData" refetching on the client. Here's what it looks like:

Page loads with the data, I can see it in `window.__NUXT__`, but then it refetches in the client so I see a flash of content.

![Screen Recording 2023-07-06 at 11 54 04](https://github.com/nuxt/nuxt/assets/91064566/16e5dd0f-0ff9-49ed-82fe-3e4f2560ddbe)

Here's how it's fetched
```
const { data: merchantStats } = useLazyAsyncData('merchant-stats-data', () =>
  merchantStore.fetchMerchantStats(merchantUrlSlug.value),
);
````

`fetchMerchantStats` is just a method that uses a plugin that returns an instance of `$fetch`.

Any ideas? Were there any breaking changes to `useAsyncData`?

**Comment by Hebilicious** [comment]
> After updating to 3.6.2, I'm seeing "useLazyAsyncData" refetching on the client. Here's what it looks like:
> 
> Page loads with the data, I can see it in `window.__NUXT__`, but then it refetches in the client so I see a flash of content.
> 
> ![Screen Recording 2023-07-06 at 11 54 04](https://user-images.githubusercontent.com/91064566/251490254-16e5dd0f-0ff9-49ed-82fe-3e4f2560ddbe.gif)
> 
> Here's how it's fetched
> 
> ```
> const { data: merchantStats } = useLazyAsyncData('merchant-stats-data', () =>
>   merchantStore.fetchMerchantStats(merchantUrlSlug.value),
> );
> ```
> 
> `fetchMerchantStats` is just a method that uses a plugin that returns an instance of `$fetch`.
> 
> Any ideas? Were there any breaking changes to `useAsyncData`?

This could be a bug, please open an issue with a stackblitz reproduction.

**Reply by maxarias-io** [reply]
No luck so far: https://stackblitz.com/edit/github-9vybbi?file=components%2FTestComponent.vue

I'm thinking that maybe the component is being re-rendered, since `useLazyAsyncData` is inside the component maybe it's refetching because of that. I'll post here if I can fix it 👍

**Reply by Hebilicious** [reply]
Can you reproduce it with this stackblitz?

**Reply by maxarias-io** [reply]
No, that stackblitz works fine.  I'm guessing it's compounded or caused by something else, still investigating/

**Reply by maxarias-io** [reply]
So maybe it was a misunderstanding on my part, but if I use a `await useAsyncData` I see the expected behavior. I don't know if this changed recently, or maybe it was just luck that I never saw issues. 

What's confusing to me is that the page does come rendered with the data with `useLazyAsyncData`, but then it refetches in the client? Is that intended?
